### PR TITLE
Add Ghana holidays

### LINKED ISSUE #1191: Add Ghana holidays
Links:
  - https://en.wikipedia.org/wiki/Public_holidays_in_Ghana
  - https://www.timeanddate.com/holidays/ghana/
  - https://www.mint.gov.gh/statutory-public-holidays/
  - https://www.bog.gov.gh/official-holidays/

### PR DESCRIPTION
**Pull request by JerryAgbesi**
## Proposed change

Adds support for official public holidays observed in Ghana

closes #1191 

## Type of change

- [x] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

File: holidays/countries/ghana.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+from holidays.groups import InternationalHolidays, ChristianHolidays, IslamicHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON
+
+
+class Ghana(
+    ObservedHolidayBase,
+    InternationalHolidays,
+    ChristianHolidays,
+    IslamicHolidays,
+):
```
Comment (arkid15r) [inline holidays/countries/ghana.py:22]:
```suggestion
class Ghana(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
```

File: holidays/countries/ghana.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    https://www.mint.gov.gh/statutory-public-holidays/
+    https://en.wikipedia.org/wiki/Public_holidays_in_Ghana
+    """
+
+    country = "GH"
+    observed_label = "%s (Observed)"
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
```
Comment (arkid15r) [inline holidays/countries/ghana.py:33]: `InternationalHolidays` init is missing

File: holidays/countries/ghana.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        # Founders' Day
+        if self._year >= 2019:
+            self._add_observed(self._add_holiday_aug_4("Founders' Day"))
+
+        # Kwame Nkrumah Memorial Day (formerly founder's Day))
+        self._add_observed(self._add_holiday_sep_21("Kwame Nkrumah Memorial Day"))
+
+        # Farmer's Day: First Friday of December
+        self._add_observed(
+            self._add_holiday("Farmer's Day", _get_nth_weekday_of_month(1, FRI, DEC, self._year))
```
Comment (arkid15r) [inline holidays/countries/ghana.py:76]: You can avoid `holidays.calendars.gregorian` imports by using
```suggestion
            self._add_observed(self._add_holiday_1st_fri_of_dec("Farmer's Day"))
```

File: tests/countries/test_ghana.py
Comment (arkid15r) [inline tests/countries/test_ghana.py:1]: Thanks a lot for adding the test! Could you include a full holiday list test (`self.assertHolidays`) for at least one year (see [example](https://github.com/JerryAgbesi/python-holidays/blob/a879260117e467abd365a1a0bc0bac968d3efbec/tests/countries/test_ukraine.py#L412))?

**Review by arkid15r — COMMENTED** [review]
@JerryAgbesi thank you for implementing a new country support. The first one in 2024!

Please take a look at some improvement suggestions when you have a chance:

File: holidays/countries/ghana.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        )
+
+        # Christmas Day
+        self._add_observed(self._add_christmas_day("Christmas Day"))
```
Comment (KJhellico) [inline holidays/countries/ghana.py:80]:
```python
        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SUN_TO_NEXT_TUE)
```
if observed only from Sunday, or
```python
        # Christmas Day
        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)

        # Boxing Day
        self._add_observed(self._add_christmas_day_two("Boxing Day"), rule=SAT_SUN_TO_NEXT_MON_TUE)
```
if observed only from Saturday and Sunday.

**Review by KJhellico — COMMENTED** [review]
@JerryAgbesi, new countries are always welcomed! 👍 Please take a look at a few suggestions.

**Comment by JerryAgbesi** [comment]
Alright great, I'll add in the changes. Cheers 🥂

File: holidays/countries/ghana.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/dr-prodigy/python-holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.groups import InternationalHolidays, ChristianHolidays, IslamicHolidays
```
Comment (arkid15r) [inline holidays/countries/ghana.py:12]:
```suggestion
from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
```

File: holidays/countries/ghana.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class Ghana(ObservedHolidayBase, InternationalHolidays, ChristianHolidays, IslamicHolidays):
```
Comment (arkid15r) [inline holidays/countries/ghana.py:20]:
```suggestion
class Ghana(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
```

File: holidays/countries/ghana.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    https://en.wikipedia.org/wiki/Public_holidays_in_Ghana
+    """
+
+    country = "GH"
+    observed_label = "%s (observed)"
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
```
Comment (arkid15r) [inline holidays/countries/ghana.py:32]:
```suggestion
        ChristianHolidays.__init__(self)
        InternationalHolidays.__init__(self)
        IslamicHolidays.__init__(self)
```

File: tests/countries/test_ghana.py [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+            ("2024-01-01", "New Year's Day"),
+            ("2024-01-07", "Constitution Day"),
+            ("2024-01-08", "Constitution Day (observed)"),
+            ("2024-03-06", "Independence Day"),
+            ("2024-03-29", "Good Friday"),
+            ("2024-04-01", "Easter Monday"),
+            ("2024-05-01", "May Day"),
+            ("2024-04-10", "Eid ul-Fitr (estimated)"),
+            ("2024-06-16", "Eid ul-Adha (estimated)"),
+            ("2024-06-17", "Eid ul-Adha (estimated) (observed)"),
```
Comment (arkid15r) [inline tests/countries/test_ghana.py:98]: We've recently changed this a bit. So now it'd make sense to add `estimated_label` and `observed_estimated_label` (see, for example, [Brunei](https://github.com/JerryAgbesi/python-holidays/blob/c1d71cd6b72924d2e4984335e4abb1d71211a399/holidays/countries/brunei.py#L75)). Please note, you don't have to add l10n wrapper `tr(` unless you want/able to implement Akan l10n (that would be super cool for this or any other PR).

Comment (JerryAgbesi) [reply tests/countries/test_ghana.py:98]: Implementing Akan translations would be quite difficult but I'll check it out a raise an issue if necessary.

Comment (arkid15r) [reply tests/countries/test_ghana.py]: > Implementing Akan translations would be quite difficult but I'll check it out a raise an issue if necessary.

That's completely optional. Thanks for considering this!

File: holidays/countries/ghana.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+class Ghana(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """
+    https://www.mint.gov.gh/statutory-public-holidays/
+    https://en.wikipedia.org/wiki/Public_holidays_in_Ghana
+    """
+
+    country = "GH"
+    estimated_label = "%s (estimated)"
+    observed_label = "%s (observed)"
+    observed_estimated_label = "%s (estimated) (observed)"
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/countries/ghana.py]: Fixed that.

**Review by arkid15r — COMMENTED** [review]
@KJhellico I hope we can finally accept it?

**Review by KJhellico — APPROVED** [review]
LGTM!

**Pull request merged**